docs/nddata/utils.rst

```diff
@@ -0,0 +1,257 @@
+the cutout is not fully contained in the data array::
+
+    >>> c2 = Cutout(data2, (0, 0), (3, 3), mode='partial')
+    >>> print(c2.data)
+    [[ nan  nan  nan]
+     [ nan   0.   1.]
+     [ nan   4.   5.]]
+
+Note that for the ``'partial'`` mode, the positions (and several other
+attributes) are calculated for on the *valid* (non-filled) cutout
+values::
+
+    >>> print(c2.position_large, c2.position_small)
+    ((0, 0), (1, 1))
+    >>> print(c2.origin_large, c2.origin_small)
+    ((0, 0), (1, 1))
+    >>> print(c2.slices_large)
+    (slice(0, 2, None), slice(0, 2, None))
+    >>> print(c2.slices_small)
+    (slice(1, 3, None), slice(1, 3, None))
+
+Using ``mode='strict'`` will raise an exception if the cutout is not
+fully contained in the data array:
+
+.. doctest-skip::
+
+    >>> c3 = Cutout(data2, (0, 0), (3, 3), mode='strict')
+    PartialOverlapError: Arrays overlap only partially.
+
+
+Cutout from a `~astropy.coordinates.SkyCoord` position
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The input ``position`` can also be specified as a
+`~astropy.coordinates.SkyCoord`, in which case a `~astropy.wcs.WCS`
+object must be input via the ``wcs`` keyword.
+
+First, let's define a `~astropy.coordinates.SkyCoord` position and a
+`~astropy.wcs.WCS` object for our data (usually this would come from
+your FITS header)::
+
+    >>> from astropy.coordinates import SkyCoord
+    >>> from astropy.wcs import WCS
+    >>> position = SkyCoord('13h11m29.96s -01d19m18.7s', frame='icrs')
+    >>> wcs = WCS(naxis=2)
+    >>> rho = np.pi / 3.
+    >>> scale = 0.05 / 3600.
+    >>> wcs.wcs.cd = [[scale*np.cos(rho), -scale*np.sin(rho)],
+    ...               [scale*np.sin(rho), scale*np.cos(rho)]]
+    >>> wcs.wcs.ctype = ['RA---TAN', 'DEC--TAN']
+    >>> wcs.wcs.crval = [position.ra.value, position.dec.value]
+    >>> wcs.wcs.crpix = [100, 50]
+
+Now let's create the cutout array using the
+`~astropy.coordinates.SkyCoord` position and ``wcs`` object::
+
+    >>> cutout = Cutout(data, position, (30, 40), wcs=wcs)
+    >>> plt.imshow(cutout.data, origin='lower')   # doctest: +SKIP
+
+.. plot::
+
+    import numpy as np
+    import matplotlib.pyplot as plt
+    from astropy.modeling.models import Gaussian2D
+    from astropy.nddata import Cutout
+    from astropy.coordinates import SkyCoord
+    from astropy.wcs import WCS
+    y, x = np.mgrid[0:500, 0:500]
+    data = Gaussian2D(1, 50, 100, 10, 5, theta=0.5)(x, y)
+    position = SkyCoord('13h11m29.96s -01d19m18.7s', frame='icrs')
+    shape = (40, 50)
+    wcs = WCS(naxis=2)
+    rho = np.pi / 3.
+    scale = 0.05 / 3600.
+    wcs.wcs.cd = [[scale*np.cos(rho), -scale*np.sin(rho)],
+                  [scale*np.sin(rho), scale*np.cos(rho)]]
+    wcs.wcs.ctype = ['RA---TAN', 'DEC--TAN']
+    wcs.wcs.crval = [position.ra.value, position.dec.value]
+    wcs.wcs.crpix = [100, 50]
+    cutout = Cutout(data, position, (30, 40), wcs=wcs)
+    plt.imshow(cutout.data)
```

Comment:
Add `origin='lower'` here (currently the image is flipped in y wrt. rest of the tutorial).